src/BuildingRegistry/Building/Building_BuildingUnit.cs

```diff
@@ -218,6 +217,14 @@ public void RemoveBuildingUnit(BuildingUnitPersistentLocalId buildingUnitPersist
 
             buildingUnit.Remove();
 
+            if (_buildingUnits.HasCommonBuildingUnit()
+                && !_buildingUnits.NotCommonBuildingUnits().Any())
```

Comment:
RegularBuildingUnits? This was kind of hard to read with fresh eyes.

Reply:
When I read it first time, I thought `NonCommonBuildingUnits` would be a bit better than `Not`. But since it was so small I didn't mind.